### DISCUSSION TITLE
Tip: Use docker images for pyspark and postgres for testing

### DISCUSSION THREAD

**Discussion by NickCrews** [moj-analytical-services/splink#2326] [Ideas]
[Ibis uses docker and docker-compose to host the various backends.](https://github.com/ibis-project/ibis/blob/17be43a1081c34582408d4c627191263ec948ada/compose.yaml) I thought this was a great idea and wanted to point it out.

Doing this would
- pro: make it easier to have repeatable dev environments
- pro: easier to resolve python dependencies. Now you don't have to bring in all the transitive python deps for `pyspark` or `psyopg2`!
- con: require devs/CI to have docker
- con: [maybe warrant some more scripting to admin this stuff](https://github.com/ibis-project/ibis/blob/17be43a1081c34582408d4c627191263ec948ada/justfile#L100)

**Comment by NickCrews** [comment]
This lib looks like it makes this trivial: https://testcontainers-python.readthedocs.io/en/latest/README.html

**Comment by RobinL** [comment]
Thanks - sounds like a good idea to me too. Just need to find the time to do it!

It's interesting they don't seem to use docker for Spark , at least I can't see it in the docker compose.  But we certainly could
